### DISCUSSION TITLE
Toast Push

### DISCUSSION THREAD

**Discussion by IsaacHatilima** [usernotnull/tall-toasts#538] [Q&A] [answered]
How do the code below work?

   ```
 toast()
        ->success('You earned a cookie! 🍪')
        ->push();
```
My expectation is that when the code reaches there is toast shows but only works with page reload. And if it requires page reload then the Livewire in the title is overkill.

**Comment by usernotnull** [comment]
> How do the code below work?
> 
>    ```
>  toast()
>         ->success('You earned a cookie! 🍪')
>         ->push();
> ```
> My expectation is that when the code reaches there is toast shows but only works with page reload. And if it requires page reload then the Livewire in the title is overkill.

There is no need for a refresh, of course.
It seems there's an issue somewhere in your implementation.
Try pushOnNextPage maybe for now?

**Reply by IsaacHatilima** [reply]
I am not redirecting after logic implementation. I just clear my form and then need to show the toast. What I did

1. composer install
2. added the code snipet to tailwind.conf file and app.js

I have a wire:click method that does only one thing, show the toast but toast only shows after page reload. I have

- Livewire : v3.5.0
- Laravel : v11.8.0

**Comment by usernotnull** [comment, marked as answer]
To properly dispatch toasts from inside your livewire components, add the trait:

use Livewire\Component;
use Usernotnull\Toast\Concerns\WireToast;

class DemoComponent extends Component
{
    use WireToast; // <-- add this

Did you add it?

**Comment by IsaacHatilima** [comment]
My bad, I left that part out. Works perfectly now. Thank you for the quick assist.
